src/CompletedTaskList.js

```diff
@@ -2,12 +2,46 @@ import React from "react"
 
 import CreatedTask from "./CreatedTask"
 import Header from "./Header"
+import todosData from "./todosData"
 
-export default function CompletedTaskList(props) {
-  return (
-    <div>
-      <Header title="Completed Tasks" />
-      <CreatedTask task={props.task} completed="completed" />
-    </div>
-  )
-}
\ No newline at end of file
+class CompletedTaskList extends React.Component {
```

Comment:
I see that the original component was a [Functional Component](https://javascriptplayground.com/functional-stateless-components-react/). I've had a lot more success using these SFC and [hooks](https://reactjs.org/docs/hooks-intro.html) instead of class components. It's a bit to wrap your head around, but I have found it makes my classes much less complex. Also if you ever switch to typescript it will be much easier to infer types.

Reply:
I agree with this! My comments below are helpful if you stick with a class component but a `Stateless Functional Component` using `hooks`, specifically `setState` would make this less complex.

Reply:
@knittingarch this is the new stuff post-2017 I was talking about! 😸 I'll probably only be able to give feedback in the class / functional component structure.

Reply:
@edwardloveall I've seen SFC in the wild!  I didn't even know that's what they were...  Woot!  Thanks for alerting me to this new option!